### PR TITLE
Update dependencies with update_deps.sh; update code due to changed libraries

### PR DESCRIPTION
**Pull request by jacob-delgado**
Run BUILD_WITH_CONTAINERS=1 UPDATE_BRANCH=master ./bin/update_deps.sh; make gen

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by jacob-delgado** [comment]
I've put a DNM/hold on this until we can get resolution. This may alter behavior that we do not want.

cc @therealmitchconnors I've been updating dependencies getting ready for 1.9 tomorrow. I was able to track some of the changes made due to the ones you did in the ledger for https://github.com/istio/pkg/pull/278. Please take a look over the PR. We should come to a resolution that is satisfactory to Networking and UX.

**Comment by jacob-delgado** [comment]
cc @howardjohn I know you approved of the initial PR. But updates have been made.

File: pilot/pkg/bootstrap/util.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func hasKubeRegistry(registries []string) bool {
 func buildLedger(ca RegistryOptions) ledger.Ledger {
 	var result ledger.Ledger
 	if ca.DistributionTrackingEnabled {
-		result = ledger.Make(ca.DistributionCacheRetention)
```
Comment (jacob-delgado) [inline pilot/pkg/bootstrap/util.go:35]: This was changed in https://github.com/istio/pkg/pull/278

File: pilot/pkg/status/ledger.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func tryLedgerPut(configLedger ledger.Ledger, obj config.Config) {
 
 func tryLedgerDelete(configLedger ledger.Ledger, obj config.Config) {
 	key := config.Key(obj.GroupVersionKind.Kind, obj.Name, obj.Namespace)
-	if err := configLedger.Delete(key); err != nil {
+	if _, err := configLedger.Delete(key); err != nil {
```
Comment (jacob-delgado) [inline pilot/pkg/status/ledger.go:31]: The `Delete` returns a `result` as well an error. Should we do anything with the result? It's a string.

**Review by ericvn — APPROVED** [review]

**Pull request merged**